### PR TITLE
Change download logic to not completely abort in certain delegation failure cases

### PR DESCRIPTION
**Pull request by cyli**
For instance if there aren't enough signatures, or if the metadata is expired

### CODE REVIEW

File: client/client_update_test.go [hunk 1]
```diff
@@ -1087,6 +1099,14 @@ func TestUpdateNonRootRemoteCorruptedCannotUseLocalCache(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping test in short mode")
 	}
+
+	acceptDelgAnyway := map[string]struct{}{
```
Comment (riyazdf) [inline client/client_update_test.go]: I think it might be worth factoring this out into a global var like `waysToMessUpServer`

**Comment by endophage** [comment]
LGTM

File: client/client_update_test.go [hunk 2]
```diff
@@ -1078,6 +1104,22 @@ func TestUpdateNonRootRemoteCorruptedCanUseLocalCache(t *testing.T) {
 	}
 }
 
+// requires that a delegation role and its decendents were not accepted as a valid part of the
```
Comment (riyazdf) [inline client/client_update_test.go]: typo: `descendants`

File: client/client_update_test.go [hunk 3]
```diff
@@ -1087,22 +1129,51 @@ func TestUpdateNonRootRemoteCorruptedCannotUseLocalCache(t *testing.T) {
...
-			// signatures.
-			shouldFail := role != data.CanonicalTimestampRole || testData.desc == "insufficient signatures"
-			testUpdateRemoteCorruptValidChecksum(t, updateOpts{
-				serverHasNewData: true,
-				localCache:       true,
-				role:             role,
-			}, testData, shouldFail)
+		case data.CanonicalTimestampRole:
+			for _, testData := range waysToMessUpServer {
+				// in general the cached timsestamp will always succeed, but if the threshold has been
```
Comment (riyazdf) [inline client/client_update_test.go]: typo: `timestamp`

**Comment by riyazdf** [comment]
LGTM pending CI - documentation CI errors are unrelated

**Pull request merged**